packages/flutter/lib/src/material/user_accounts_drawer_header.dart

```diff
@@ -0,0 +1,165 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+import 'debug.dart';
+
+/// The top-most region of a material design drawer with user accounts. The
+/// header's [decoration] is used to provide a background.
+/// [currentAccountPicture] is the main account picture on the left, while
+/// [otherAccountsPictures] are the smaller account pictures on the right.
+/// [accountName] and [accountEmail] provide access tothe top and bottom rows
+/// of the account details in the lower part of the header. When touched, this
+/// area triggers [onDetailsPressed] and toggles the dropdown icon on the right.
+///
+/// Part of the material design [Drawer].
```

Comment:
We usually put a summary line like this first.

You might make the summary a little more specific: A material design [Drawer] header that identifies that app's user.